src/TLuaInterpreter.cpp

```diff
@@ -3029,51 +3029,96 @@ int TLuaInterpreter::setMainWindowSize(lua_State* L)
 
 int TLuaInterpreter::setBackgroundColor(lua_State* L)
 {
-    string luaSendText = "";
-    if (!lua_isstring(L, 1)) {
-        lua_pushstring(L, "setBackgroundColor: wrong argument type");
-        lua_error(L);
-        return 1;
-    } else {
-        luaSendText = lua_tostring(L, 1);
-    }
-    double x1;
-    if (!lua_isnumber(L, 2)) {
-        lua_pushstring(L, "setBackgroundColor: wrong argument type");
-        lua_error(L);
-        return 1;
+    Host* pHost = &getHostFromLua(L);
+    QString windowName;
+    int r, g, b, alpha;
+
+    auto validRange = [](int number) {
```

Comment:
💡  What about other functions, range checking could be required in other places in this `class`, so why not just make an inclusive range checker we can in-line where needed:
```
static bool TLuaInterpreter::validRange(const int& value, const int& lowerLimit, const int& upperLimit)
{
    return ((number >= lowerLimit) && (number <=upperLimit));
}
```

Reply:
If we'll need it, we'll move it, no need to clutter the class interface otherwise.